src/app/ssh-keys/redux/ssh-key.reducers.ts

```diff
@@ -0,0 +1,203 @@
+import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';
+import { SSHKeyPair } from '../../shared/models/ssh-keypair.model';
+import {
+  combineReducers,
+  compose,
+  createFeatureSelector,
+  createSelector
+} from '@ngrx/store';
+import { Account } from '../../shared/models/account.model';
+
+import * as sshKey from './ssh-key.actions';
+
+export interface State {
+  list: ListState,
+  form: FormState
+}
+
+export interface ListState extends EntityState<SSHKeyPair> {
+  filters: {
+    selectedGroupings: any[],
+    selectedAccounts: Account[]
+  }
+}
+
+const initialListState: ListState = {
+  ids: [],
+  entities: null,
+  filters: {
+    selectedAccounts: [],
+    selectedGroupings: []
+  }
+};
+
+export interface FormState {
+  loading: boolean,
+  error: Object
+}
+
+const initialFormState: FormState = {
+  loading: false,
+  error: null
+};
+
+
+export interface SshKeysState {
+  list: ListState;
+  form: FormState;
+}
+
+export const sshKeyReducers = compose(combineReducers)({
+  list: listReducer,
+  form: formReducer
+});
+
+export const adapter: EntityAdapter<SSHKeyPair> = createEntityAdapter<SSHKeyPair>({
+  selectId: (item: SSHKeyPair) => `${item.account}-${item.name}`,
```

Comment:
`${item.account}-${item.name}` is not unique. I think it's better to add domainId and move this to a separate function to use in reducer for generating the key